include/verilated_types.h

```diff
@@ -516,7 +516,7 @@ class VlQueue final {
     VlQueue<IData> unique_index(Func with_func) const {
         VlQueue<IData> out;
         IData index = 0;
-        std::unordered_set<T_Value> saw;
+        std::unordered_set<decltype(with_func(0, m_deque[0]))> saw;
```

Comment:
This probably should have always been std::set for better performance.

Perhaps use a 'using' typedef in place of the two identical decltypes?

With those feel free to squash merge after tests pass.

Thanks!

Reply:
I changed the type of the set.
But I don't know if I can use `using`. Although the expressions of dectlypes are the same, these `Func` are different, so decltypes may be resolved to different types.